publishing-bot: Use go1.15.0 as default version and go1.13.15 for previous releases

justaugustus opened the pull request:
**What type of PR is this?**

/kind cleanup
/area dependency

**What this PR does / why we need it**:

- publishing-bot: Use go1.15.0 as default version
- publishing-bot: Use go1.13.15 for 1.18, 1.17, and 1.16 release branches

Tracking issues: https://github.com/kubernetes/release/issues/1421, https://github.com/kubernetes/release/issues/1474
Closes: https://github.com/kubernetes/release/issues/1474

/assign @nikhita @liggitt
cc: @kubernetes/release-engineering @kubernetes/publishing-bot-reviewers 

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

justaugustus committed:
publishing: Use go1.13.15 for 1.18, 1.17, and 1.16 release branches

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

justaugustus committed:
publishing: Use go1.15.0 as default version

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

nikhita commented on staging/publishing/rules.yaml:
```diff
@@ -4,7 +4,7 @@ recursive-delete-patterns:
 - "*/BUILD.bazel"
 - Gopkg.toml
-default-go-version: 1.14.6
+default-go-version: 1.15.0
```
```suggestion
default-go-version: 1.15
```

nikhita replied:
Would need to do this for all 1.15 versions below

justaugustus replied:
Oh geez, really?
_shakes fist at go minor versions_

justaugustus replied:
Fixed.

justaugustus commented:
/priority critical-urgent
/milestone v1.19

nikhita commented on staging/publishing/rules.yaml:
```diff
@@ -1640,9 +1640,9 @@ rules:
       dir: staging/src/k8s.io/controller-manager
     name: master
-    go: 1.13.9
+    go: 1.13.15
```
master would be using go 1.15 so we'd need to remove this line (was incorrect before)

justaugustus replied:
Fixed.

justaugustus committed:
publishing: Drop 'go' field from 'controller-manager' 'master' branch

This was previous incorrectly set to go1.13.
Dropping the field allows us to pick up the 'default-go-version'.

Signed-off-by: Stephen Augustus <saugustus@vmware.com>

nikhita commented:
Thanks, Stephen!

/lgtm
/approve

justaugustus commented:
/test pull-kubernetes-e2e-gce-100-performance

nikhita commented:
/retest

On Fri, Aug 14, 2020 at 5:06 PM Kubernetes Prow Robot <
notifications@github.com> wrote:

> @justaugustus <https://github.com/justaugustus>: The following test
> *failed*, say /retest to rerun all failed tests:
> Test name Commit Details Rerun command
> pull-kubernetes-e2e-gce-100-performance 7681a3a
> <https://github.com/kubernetes/kubernetes/commit/7681a3a0db3fc0315e3c9adcb72bd66c2cb9df21>
> link
> <https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/93998/pull-kubernetes-e2e-gce-100-performance/1294219543576580096/> /test
> pull-kubernetes-e2e-gce-100-performance
>
> Full PR test history
> <https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=93998>. Your
> PR dashboard. Please help us cut down on flakes by linking to
> <https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests>
> an open issue
> <https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open> when
> you hit one in your PR.
>
> Instructions for interacting with me using PR comments are available here
> <https://git.k8s.io/community/contributors/guide/pull-requests.md>. If
> you have questions or suggestions related to my behavior, please file an
> issue against the kubernetes/test-infra
> <https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:>
> repository. I understand the commands that are listed here
> <https://go.k8s.io/bot-commands>.
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/93998#issuecomment-674032291>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD24BUHQU2BR7XT4BK5D5RDSAUOSTANCNFSM4P7JE57Q>
> .
>
